src/features/governance/components/Details.tsx

```diff
@@ -47,23 +47,23 @@ export const Details = ({ proposal }: Props) => {
   }, []);
 
   const pastVote = useMemo(() => {
-    return getProposalVote(proposal.proposalID.toString());
-  }, [getProposalVote, proposal.proposalID]);
+    return getProposalVote(proposal.proposalID.toString(), address!);
+  }, [getProposalVote, proposal.proposalID, address]);
 
   const onVote = useCallback(async () => {
     if (!currentVote || hasVoted) return;
     setTransactionModalOpen(true);
     await voteProposal(proposal, currentVote, { onSent: () => setTransactionModalOpen(false) });
-    setHasVoted(true);
   }, [currentVote, hasVoted, voteProposal, proposal]);
 
   useEffect(() => {
+    console.info('Checking if user has voted', address, proposal.proposalID);
```

Comment:
hack but i added this in so the hook uses address and thus def will run when it changes

Reply:
99.9% sure this isn't an issue as `getHasVoted` is `useAsyncCallback` wrapping the result of a `useContractRead` using `address` already. This will provoke `getHasVoted` to change